async-nats/src/jetstream/context.rs

```diff
@@ -105,6 +107,50 @@ impl Context {
         }
     }
 
+    /// Publishes [Message] to the [crate::jetstream::stream::Stream] without waiting for
+    /// acknowledgment from the server that the message has been successfully delivered.
+    ///
+    /// Acknowledgment future that can be polled is returned instead.
+    ///
+    /// If the stream does not exist, `no responders` error will be returned.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<(), async_nats::Error> {
+    /// let client = async_nats::connect("localhost:4222").await?;
+    /// let jetstream = async_nats::jetstream::new(client);
+    ///
+    /// let first_ack = jetstream.publish_async("events".to_string(), "data".into()).await?;
+    /// let second_ack = jetstream.publish_async("events".to_string(), "data".into()).await?;
+    /// first_ack.await?;
+    /// second_ack.await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn publish_async(
```

Comment:
Few thoughts:
* All other clients have explicit notion of "do not wait for ack" method.
* this will not break the code for users who users who just do `context.publish().await?`. A big surprise.
BUT
`publish_async` looks very ugly.

Probably a compromise is to have a better name. shorten version of... `publish_dont_wait_for_ack`

Reply:
Let's go with one method only, the one not waiting or ack by default.

Reply:
Naming wise.. something like `publish_with_deferred_ack`, `publush_without_waiting_for_ack`, `publish_no_wait_ack` etc?

Reply:
Well, if we go with `publish` being not a waiting one, we don't need to think about above :)